features/DILL/Claim.tsx

```diff
@@ -0,0 +1,160 @@
+import { useState, FC, useEffect } from "react";
+import { Spacer, Grid, Card, Button } from "@geist-ui/react";
+import styled from "styled-components";
+import { formatEther } from "ethers/lib/utils";
+import { Contracts } from "../../containers/Contracts";
+import { Connection } from "../../containers/Connection";
+import { Dill, UseDillOutput } from "../../containers/Dill";
+import {
+  ERC20Transfer,
+  Status as ERC20TransferStatus,
+} from "../../containers/Erc20Transfer";
+
+interface ButtonStatus {
+  disabled: boolean;
+  text: string;
+}
+
+const setButtonStatus = (
+  status: ERC20TransferStatus,
+  transfering: string,
+  idle: string,
+  setButtonText: (arg0: ButtonStatus) => void,
+) => {
+  if (status === ERC20TransferStatus.Transfering) {
+    setButtonText({
+      disabled: true,
+      text: transfering,
+    });
+  }
+
+  if (
+    status === ERC20TransferStatus.Success ||
+    status === ERC20TransferStatus.Failed ||
+    status === ERC20TransferStatus.Cancelled
+  ) {
+    setButtonText({
+      disabled: false,
+      text: idle,
+    });
+  }
+};
+
+const formatNumber = (num: number, precision?: number) =>
+  num &&
+  (num.toLocaleString(undefined, {
+    minimumFractionDigits: precision || 2,
+    maximumFractionDigits: precision || 2,
+  }) ||
+    0);
+
+const formatPercent = (decimal: number) =>
+  decimal &&
+  (decimal * 100).toLocaleString(undefined, {
+    minimumFractionDigits: 2,
+    maximumFractionDigits: 2,
+  }) + "%";
+
+export const Claim: FC<{
+  dillStats: UseDillOutput;
+}> = ({ dillStats }) => {
+  const { blockNum, address, signer } = Connection.useContainer();
+  const { feeDistributor } = Contracts.useContainer();
+  const [claimable, setClaimable] = useState<number | null>(null);
+
+  const [claimButton, setClaimButton] = useState<ButtonStatus>({
+    disabled: claimable ? false : true,
+    text: `Claim ${claimable ? formatNumber(claimable, 3) : "0"} PICKLEs`,
+  });
+  const {
+    status: transferStatus,
+    transfer,
+    getTransferStatus,
+  } = ERC20Transfer.useContainer();
+
+  const dillAPY =
+    dillStats.weeklyDistribution && dillStats.lockedValue
+      ? dillStats.weeklyDistribution / dillStats.lockedValue
```

Comment:
This appears to be calculating the amount of return per week. multiply by 52 to get apy. #
```suggestion
      ? dillStats.weeklyDistribution / dillStats.lockedValue * 52
```

Reply:
If you want it more accurate you can multiply by 52.17857